src/core/client_channel/subchannel.cc

```diff
@@ -422,10 +422,17 @@ class Subchannel::ConnectedSubchannelStateWatcher final
         if (c->channelz_node() != nullptr) {
           c->channelz_node()->SetChildSocket(nullptr);
         }
-        // Even though we're reporting IDLE instead of TRANSIENT_FAILURE here,
-        // pass along the status from the transport, since it may have
-        // keepalive info attached to it that the channel needs.
-        // TODO(roth): Consider whether there's a cleaner way to do this.
+        // If the subchannel was created from an endpoint we report
```

Comment:
I think some of the details are confused here -- for example, the keepalive propagation happens when we report IDLE, not TRANSIENT_FAILURE.  I suggest the following wording instead:

"""
If the subchannel was created from an endpoint, then we report TRANSIENT_FAILURE here instead of IDLE.  The subchannel will never leave TRANSIENT_FAILURE state, because there is no way for us to establish a new connection.

Otherwise, we report IDLE here.  Note that even though we're not reporting TRANSIENT_FAILURE, we pass along the status from the transport, since it may have keepalive info attached to it that the channel needs.
TODO(roth): Consider whether there's a cleaner way to propagate the keepalive info.
"""